src/charm.py

```diff
@@ -1280,6 +1280,17 @@ def _on_update_status(self, _) -> None:
             logger.debug("on_update_status early exit: Service has not been added nor started yet")
             return
 
+        if (
+            "restoring-backup" not in self.app_peer_data
```

Comment:
I believe async replication may also need to skip this logic.

https://github.com/canonical/postgresql-k8s-operator/blob/ea071d617bde602e0b168496b29e3758e3a47a63/src/relations/async_replication.py#L704

One way to check it is by checking whether the `stopped` flag is set in the relation data. It's being set after the container is stopped.

https://github.com/canonical/postgresql-k8s-operator/blob/ea071d617bde602e0b168496b29e3758e3a47a63/src/relations/async_replication.py#L718

Reply:
Honestly surprised this didn't cause the async_replication test to fail. In any case, nice catch, I'll add this check